lib/client/keystore.go

```diff
@@ -160,17 +160,19 @@ func (fs *FSKeyStore) AddKey(key *Key) error {
 	if err := fs.writeBytes(key.TLSCert, fs.tlsCertPath(key.KeyIndex)); err != nil {
 		return trace.Wrap(err)
 	}
+
+	// We only generate PPK files for use by PuTTY when running tsh on Windows.
 	if runtime.GOOS == constants.WindowsOS {
 		ppkFile, err := key.PPKFile()
-		if err == nil {
+		// PPKFile can only be generated from an RSA private key. If the key is in a different
+		// format, a BadParameter error is returned and we can skip PPK generation.
+		if err != nil && !trace.IsBadParameter(err) {
+			fs.log.WithError(err).Debugf("Cannot convert private key to PPK-formatted keypair.")
```

Comment:
`BadParameter` is thrown when the key passed in cannot be parsed as `rsa.PrivateKey`: https://github.com/gravitational/teleport/blob/6973cd652720011c2fb6466cfe0889533d9dfae8/api/utils/keys/privatekey.go#L137-L140

We just ignore this key in that case and continue login as normal.

Philosophically, I've always treated failure to generate a PPK file in the first place as a non-error, because PuTTY support is pretty optional and it shouldn't block the functionality of `tsh login`. On the other hand, if we _do_ successfully generate the PPK but then fail to write it to disk successfully, I think that's a different kind of error that's worth halting a login for.